library/errors.c

```diff
@@ -1,40 +1,66 @@
-/*
-Copyright 2019-2020 NetFoundry, Inc.
-
-Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
-You may obtain a copy of the License at
-
-https://www.apache.org/licenses/LICENSE-2.0
-
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
-*/
+// Copyright (c) 2023.  NetFoundry Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
 
 #include "ziti/errors.h"
 
-enum _zt_err {
-#define _err_enum(e,_) _ziti_##e,
-    ZITI_ERRORS(_err_enum)
-};
+#define ZITI_ERRORS(XX) \
+    XX(OK, "OK") \
+    XX(CONFIG_NOT_FOUND, "Configuration not found") \
+    XX(JWT_NOT_FOUND, "JWT not found") \
+    XX(JWT_INVALID, "JWT not accepted by controller") \
+    XX(JWT_INVALID_FORMAT, "JWT has invalid format") \
+    XX(PKCS7_ASN1_PARSING_FAILED, "PKCS7/ASN.1 parsing failed") \
+    XX(JWT_SIGNING_ALG_UNSUPPORTED, "unsupported JWT signing algorithm") \
+    XX(JWT_VERIFICATION_FAILED, "JWT verification failed") \
+    XX(ENROLLMENT_METHOD_UNSUPPORTED, "unsupported enrollment method") \
+    XX(ENROLLMENT_CERTIFICATE_REQUIRED, "enrollment method requires certificate") \
+    XX(KEY_GENERATION_FAILED, "error generating private key") \
+    XX(KEY_LOAD_FAILED, "error loading TLS key") \
+    XX(CSR_GENERATION_FAILED, "error generating a CSR") \
+    XX(INVALID_CONFIG, "Configuration is invalid") \
+    XX(NOT_AUTHORIZED, "Not Authorized") \
+    XX(CONTROLLER_UNAVAILABLE, "Ziti Controller is not available") \
+    XX(GATEWAY_UNAVAILABLE, "Ziti Edge Router is not available") \
+    XX(SERVICE_UNAVAILABLE, "Service not available") \
+    XX(EOF, "Connection closed") \
```

Comment:
"by peer" might be helpful, to distinguish from `CONN_CLOSED`?

Reply:
it is kind of misleading, it just indicates that peer won't send any more data